### PR TITLE
define new keyboard "angel17"

### PR DESCRIPTION
**Pull request by kakunpc**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/angel17/rules.mk [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency.
```
Comment (noroadsleft) [inline keyboards/angel17/rules.mk]: Lines 4-40 here are actually completely unnecessary as QMK now defines defaults for those rules based on the defined MCU.

https://github.com/qmk/qmk_firmware/blob/ee8d9bd40ad4f5b8d23be951d652143582d42938/quantum/mcu_selection.mk#L39-L75

See: #6253

**Comment by kakunpc** [comment]
I fixed it.

File: keyboards/angel17/rules.mk [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
+BOOTLOADER = atmel-dfu
```
Comment (zvecr) [inline keyboards/angel17/rules.mk]: The readme mentions
>Hardware Supported: angel17_alpha, promicro

If so, the bootloader should be updated to match.

```suggestion
BOOTLOADER = caterina
```

File: keyboards/angel17/angel17.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/angel17/angel17.h]:
```suggestion
#define LAYOUT_numpad_5x4( \
```

File: keyboards/angel17/angel17.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    k00, k01, k02, k03, \
+    k10, k11, k12, k13, \
```
Comment (drashna) [inline keyboards/angel17/angel17.h]:
```suggestion
    k10, k11, k12, KC_NO, \
```

File: keyboards/angel17/angel17.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    k20, k21, k22, \
```
Comment (drashna) [inline keyboards/angel17/angel17.h]:
```suggestion
    k20, k21, k22, k13, \
```

File: keyboards/angel17/angel17.h [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    k30, k31, k32, k33, \
```
Comment (drashna) [inline keyboards/angel17/angel17.h]:
```suggestion
    k30, k31, k32, \
```

File: keyboards/angel17/angel17.h [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    k40, k41 \
```
Comment (drashna) [inline keyboards/angel17/angel17.h]:
```suggestion
    k40, k41, k33 \
```

File: keyboards/angel17/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7, KC_P8, KC_P9, KC_PPLS,
```
Comment (drashna) [inline keyboards/angel17/keymaps/default/keymap.c]:
```suggestion
    KC_P7,    KC_P8,    KC_P9,
```

File: keyboards/angel17/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    KC_P4, KC_P5, KC_P6,
```
Comment (drashna) [inline keyboards/angel17/keymaps/default/keymap.c]:
```suggestion
    KC_P4,    KC_P5,    KC_P6,    KC_PPLS,
```

File: keyboards/angel17/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
```
Comment (drashna) [inline keyboards/angel17/keymaps/default/keymap.c]:
```suggestion
  [0] = LAYOUT_numpad_5x4( /* Base */
```

File: keyboards/angel17/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7, KC_P8, KC_P9, KC_PPLS,
+    KC_P4, KC_P5, KC_P6,
+    KC_P1, KC_P2, KC_P3, KC_PENT,
```
Comment (drashna) [inline keyboards/angel17/keymaps/default/keymap.c]:
```suggestion
    KC_P1,    KC_P2,    KC_P3,
```

File: keyboards/angel17/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    KC_P0, KC_PDOT
```
Comment (drashna) [inline keyboards/angel17/keymaps/default/keymap.c]:
```suggestion
        KC_P0,    KC_PDOT,       KC_PENT
```

File: keyboards/angel17/rules.mk [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (drashna) [inline keyboards/angel17/rules.mk]:
```suggestion
HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)

LAYOUTS = numpad_5x4
```

**Comment by kakunpc** [comment]
I changed it.

File: keyboards/angel17/info.json [hunk 14]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (fauxpark) [inline keyboards/angel17/info.json]: This info.json should be arranged like so, to match the `LAYOUT` macro in angel17.h (moved the `+` and Enter positions down one row):

```json
{
    "keyboard_name": "angel17",
    "url": "https://kakunpc.booth.pm/",
    "maintainer": "kakunpc",
    "width": 4,
    "height": 5,
    "layouts": {
        "LAYOUT_numpad_5x4": {
            "layout": [
                {"label":"Num Lock", "x":0, "y":0},
                {"label":"/", "x":1, "y":0},
                {"label":"*", "x":2, "y":0},
                {"label":"-", "x":3, "y":0},

                {"label":"7", "x":0, "y":1},
                {"label":"8", "x":1, "y":1},
                {"label":"9", "x":2, "y":1},

                {"label":"4", "x":0, "y":2},
                {"label":"5", "x":1, "y":2},
                {"label":"6", "x":2, "y":2},
                {"label":"+", "x":3, "y":1, "h":2},

                {"label":"1", "x":0, "y":3},
                {"label":"2", "x":1, "y":3},
                {"label":"3", "x":2, "y":3},

                {"label":"0", "x":0, "y":4, "w":2},
                {"label":".", "x":2, "y":4},
                {"label":"Enter", "x":3, "y":3, "h":2}
            ]
        }
    }
}
```

Comment (drashna) [reply keyboards/angel17/info.json]: Ah, yeah, I didn't even think about that. 

I think that once this is done, it should be good to go.

**Comment by kakunpc** [comment]
Thank you very much for your advice.
I fixed info.json.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**